performant-software/DM
# Viewer, grid, rollover UI fixes

### What does this PR do?
- Allows image viewer elements that have been added to the grid via an annotation link to be closed whether or not they are in edit mode
- Inserts viewer elements that have been added to the grid via an annotation after the source element for that annotation rather than at end of grid
- Fixes a bug introduced in #62 that disabled click-to-edit-title feature
- Fixes a bug introduced in #76 that disabled clicks on icons in rollover window toolbar
- Retains the width of resizable rollover window when it's hidden and shown (still does not persist custom size to server)

### What issues does this address?
Addresses #58, #64, #84 

### How to test
- On the staging server (http://67.207.158.85/), add a resource to the grid that has annotations linked to it and another resource after it
- From the rollover window for an annotation, click a resource in its Targets list to add that resource to the grid as well; the newly added resource should be inserted immediately after the annotated resource, rather than at the end of the collection
- Do the same process to add an image resource to the grid via an annotation and its popup window
- Enable editing mode on that resource, then click the X to close it. The resource should be removed from the grid
- With editing mode enabled on a resource, click its title and confirm that it turns into an editable text field
- Click and drag on an empty space in the resource's title bar to confirm that the frame can be repositioned in the grid collection
- With a resource in editing mode, hover over an annotation to bring up the rollover window. Confirm that the editing icons in the rollover window's toolbar respond to clicks.
- Resize the rollover window by dragging the lower-right corner. Close the rollover window and bring it back up by hovering over another annotation; confirm that it appears to be the same size you resized it to.
